services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/structure/directory/decorator/indexing/IndexingDecoratorTest.java

```diff
@@ -73,6 +73,7 @@ void checkWritesIndexFile() throws IOException, ParseException {
     java.io.File actualIndexDirectoryFile = Objects.requireNonNull(outputFile.listFiles())[0];
     java.io.File actualPhysicalFile = Stream.of(actualIndexDirectoryFile)
         .map(File::listFiles)
+        .filter(Objects::nonNull)
```

Comment:
I was confused when I read this line -> listFiles returning null is nothing I would have expected. But when checking the docs, turns out that it does indeed return null in case the given file is not a directory - so from a readability POV, i would move the filter one line higher and check on `File::isDirectory` instead.

Reply:
There's no reference to `File::isDirectory` in the code segment in question. When applied to `Stream.of(actualIndexDirectoryFile)` (i.e. one line above) `map(File::listFiles)` might still yield `null` due to assertions in `File(String child, File parent)`.

Reply:
I was referring to line 75, which could be changed to:

```java
java.io.File actualPhysicalFile = Stream.of(actualIndexDirectoryFile)
   .filter(File::isDirectory)        
   .map(File::listFiles)
```

Reply:
Ah, I see. That makes sense. I've changed it accordingly.

Reply:
I see. That makes sense. However, this might erroneously filter out symbolic links pointing to directories because symlinks always are files, even if they refer to directories.